blocks/library/text/index.js

```diff
@@ -85,13 +91,25 @@ registerBlockType( 'core/text', {
 	},
 
 	save( { attributes } ) {
-		const { align, content } = attributes;
+		const { align, content, dropCap } = attributes;
+		const cssClassNames = classnames( {
+			'has-drop-cap': dropCap,
+		} );
+
+		// If no classes are present don't add empty attribute.
+		if ( cssClassNames.length === 0 ) {
+			if ( ! align ) {
+				return <p>{ content }</p>;
+			}
+
+			return <p style={ { textAlign: align } }>{ content }</p>;
+		}
```

Comment:
We could probably simplify the logic here to avoid duplication

`const className = dorpCap ? 'has-drop-cap' : undefined;`

And always add the className prop. (undefined is equivalent to unset)